azuread/resource_application.go

```diff
@@ -207,6 +256,10 @@ func resourceApplicationCreate(d *schema.ResourceData, meta interface{}) error {
 		RequiredResourceAccess:  expandADApplicationRequiredResourceAccess(d),
 	}
 
+	if v, ok := d.GetOk("app_role"); ok {
+		properties.AppRoles = expandADApplicationAppRoles(v.(*schema.Set).List())
```

Comment:
Basically we would change this to:
```suggestion
	AppRoles: expandADApplicationAppRoles(d.Get("app_role"))
```

and move it up into the initializer above

Reply:
sorted